common/src/com/thoughtworks/go/config/materials/git/GitMaterial.java

```diff
@@ -235,7 +235,13 @@ public void afterCompletion(int status) {
                 });
             }
             int cloneDepth = shallowClone ? preferredCloneDepth : Integer.MAX_VALUE;
-            int returnValue = gitCommand.cloneFrom(outputStreamConsumer, url.forCommandline(), cloneDepth);
+            int returnValue;
+            if(executionContext.isServer()) {
+                returnValue = gitCommand.serverSideClone(outputStreamConsumer, url.forCommandline());
```

Comment:
Hi @varshavaradarajan 
I am wondering if there is any way that checkout on server can be still executed? There is `ConfigMaterialUpdater` which relies on `MaterialPoller.checkout` to actually make a clean checkout of git repository. The path starts around [here](https://github.com/gocd/gocd/blob/master/server/src/com/thoughtworks/go/server/materials/ConfigMaterialUpdater.java#L114) and IIRC it eventually calls `updateTo` which it seems to me, now would never execute a full checkout if running on server side.